### PR TITLE
Make directory before create config file

### PR DESCRIPTION
**Pull request by nanjj**
In case of the directory does not exist.

Signed-off-by: JUN JIE NAN <nanjunjie@gmail.com>

### CODE REVIEW

**Comment by stgraber** [comment]
I don't think the lxc/config package should get into recursively creating directories.
It should be the caller's responsibility to make sure the directories exist.

**Comment by nanjj** [comment]
> I don't think the lxc/config package should get into recursively creating directories. It should be the caller's responsibility to make sure the directories exist.

fixed

File: lxc/main.go [hunk 1]
```diff
@@ -9,7 +9,7 @@ import (
 
 	"github.com/spf13/cobra"
 
-	"github.com/lxc/lxd/client"
+	lxd "github.com/lxc/lxd/client"
```
Comment (stgraber) [inline lxc/main.go:12]: This change shouldn't be included. It's most likely your editor deciding to expand the package name for some reason.

File: lxc/main.go [hunk 2]
```diff
@@ -333,7 +333,13 @@ func (c *cmdGlobal) PreRun(cmd *cobra.Command, args []string) error {
 			return err
 		}
 	} else {
-		c.conf = config.NewConfig(filepath.Dir(c.confPath), true)
+		confDir := filepath.Dir(c.confPath)
+		if !shared.PathExists(confDir) {
+			if err = os.MkdirAll(confDir, 0700); err != nil {
```
Comment (stgraber) [inline lxc/main.go:338]: LXD doesn't use the single-line if syntax in our codebase.
Please separate the assignment and check.

**Comment by stgraber** [comment]
Can you also explain exactly when this code is needed in your experience?

Looking at the logic, we seem to be actively depending on the directory NOT existing, just a few lines below.
Specifically, if the config path doesn't exist, we consider it to be LXD's first run which we then use to show a first-use message to the user.

As part of that logic, we have a call for:
```
		err = os.MkdirAll(c.conf.ConfigDir, 0750)
```

Which should have a very similar effect to the code you're adding here.

Without testing your branch, I have a strong suspicion that your change actually breaks our first-time-use logic and I'm very unclear about what it's solving given that the MkdirAll logic is already present.

**Comment by stgraber** [comment]
Closing until we understand exactly what's being fixed here.

**Pull request closed without merging**